core/auth/fake/interfaces/controller.go

```diff
@@ -0,0 +1,186 @@
+package interfaces
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"html/template"
+	"net/http"
+
+	"flamingo.me/flamingo/v3/core/auth/fake/domain"
+	"flamingo.me/flamingo/v3/framework/web"
+)
+
+type (
+	// IdpController is the fake identity provider controller
+	IdpController struct {
+		responder     *web.Responder
+		reverseRouter web.ReverseRouter
+		config        *fakeConfig
+	}
+
+	viewData struct {
+		FormURL    string
+		Message    string
+		UsernameID string
+		PasswordID string
+		OtpID      string
+	}
+)
+
+const (
+	errMissingUsername  = "missing username"
```

Comment:
`err` prefixes should be used for error variables, such as `var errMissingUsername = errors.New("missing username")`. (See https://blog.golang.org/go1.13-errors for an example of an error variable)

Reply:
Fixed - they are now errors, not strings. Thx for the notice.